cookbook/console/style.rst

```diff
@@ -0,0 +1,363 @@
+        // consider using arrays when displaying long notes
+        $io->note(array(
+            'Lorem ipsum dolor sit amet',
+            'Consectetur adipiscing elit',
+            'Aenean sit amet arcu vitae sem faucibus porta'
+        ));
+
+:method:`Symfony\\Component\\Console\\Helper\\StyleInterface::caution`
+    Similar to the ``note()`` helper, but the contents are more prominently
+    highlighted. The resulting contents resemble an error message, so you should
+    avoid using this helper unless strictly necessary::
+
+        // use simple strings for short caution message
+        $io->note('Lorem ipsum dolor sit amet');
+
+        // ...
+
+        // consider using arrays when displaying long caution messages
+        $io->note(array(
+            'Lorem ipsum dolor sit amet',
+            'Consectetur adipiscing elit',
+            'Aenean sit amet arcu vitae sem faucibus porta'
+        ));
+
+:method:`Symfony\\Component\\Console\\Helper\\StyleInterface::newLine`
+    It displays a blank line in the command output. Although it may seem useful,
+    most of the times you won't need it at all. The reason is that every helper
+    already adds their own blank lines, so you don't have to care about the
+    vertical spacing::
+
+        // outputs a single blank line
+        $io->newLine();
+
+        // outputs three consecutive blank lines
+        $io->newLine(3);
+
+:method:`Symfony\\Component\\Console\\Helper\\StyleInterface::progressStart`
+    It displays a progress bar with a number of steps equal to the argument passed
+    to the method (don't pass any value if the length of the progress bar is
+    unknown)::
+
+        // display a progress bar of unknown length
+        $io->progressStart();
+
+        // display a 100-step length progress bar
+        $io->progressStart(100);
+
+:method:`Symfony\\Component\\Console\\Helper\\StyleInterface::progressAdvance`
+    It makes the progress bar advance the given number of steps (or ``1`` step
+    if no argument is passed)::
+
+        // advanced the progress bar 1 step
+        $io->progressStart();
+
+        // advanced the progress bar 10 steps
+        $io->progressStart(10);
+
+:method:`Symfony\\Component\\Console\\Helper\\StyleInterface::progressFinish`
+    It finishes the progress bar (filling up all the remaining steps when its
+    length is known)::
+
+        $io->progressFinish(10);
+
+Asking for User's Input
+~~~~~~~~~~~~~~~~~~~~~~~
+
+:method:`Symfony\\Component\\Console\\Helper\\StyleInterface::ask`
+    It asks the user to provide some value::
+
+        $io->ask('What is your name?');
+
+    You can pass the default value as the second argument so the user can simply
+    hit the <Enter> key to select that value::
+
+        $io->ask('Where are you from?', 'United States');
+
+    In case you need to validate the given value, pass a callback validator as
+    the third argument:
+
+        $io->ask('Number of workers to start', 1, function ($number) {
```

Comment:
This block seems not to be rendered properly: http://pr-6017-6qmocelev2lwe.eu.platform.sh/cookbook/console/style.html#asking-for-user-s-input